pkg/imgpkg/image/resolved.go

```diff
@@ -31,27 +29,15 @@ func (i ResolvedImage) URL() (string, error) {
 		return "", err
 	}
 
-	imgDescriptor, err := i.imagesMetadata.Generic(tag)
+	hash, err := i.imagesMetadata.Digest(tag)
 	if err != nil {
 		return "", err
 	}
 
-	// Resolve image second time because some older registry can
-	// return "random" digests that change for every request.
-	// See https://github.com/k14s/imgpkg/issues/21 for details.
-	imgDescriptor2, err := i.imagesMetadata.Generic(tag)
-	if err != nil {
-		return "", err
-	}
-
-	if imgDescriptor.Digest.String() != imgDescriptor2.Digest.String() {
-		return "", fmt.Errorf("Expected digest resolution to be consistent over two separate requests")
-	}
-
-	digestURL, err := regname.NewDigest(tag.Repository.String() + "@" + imgDescriptor.Digest.String())
+	digest, err := regname.NewDigest(tag.Repository.String() + "@" + hash.String())
```

Comment:
it seems like ResolvedImage is only used in one place. can we inline this simplified logic there?

Reply:
Ok, i do enjoy deleting code.